packages/oslo-converter-uml-ea/lib/connector-normalisation-cases/AssociationWithNameConnectorCase.ts

```diff
@@ -0,0 +1,120 @@
+import {
+  EaConnector,
+  DataRegistry,
+  NormalizedConnector,
+  EaTag,
+} from '@oslo-flanders/ea-uml-extractor';
+import { IConnectorNormalisationCase } from '../interfaces/IConnectorNormalisationCase';
+import { inject, injectable } from 'inversify';
+import { TagNames } from '../enums/TagNames';
+import { getTagValue, toCamelCase, toPascalCase, updateNameTag } from '../utils/utils';
+import { Logger } from '@oslo-flanders/core';
+import { EaUmlConverterServiceIdentifier } from '../config/EaUmlConverterServiceIdentifier';
+
+@injectable()
+export class AssociationWithNameConnectorCase
+  implements IConnectorNormalisationCase
+{
+  @inject(EaUmlConverterServiceIdentifier.Logger)
+  public readonly logger!: Logger;
+
+  /**
+   * Case: the connector has a name and a source and/or destination cardinality.
+   *  This class does not handle the case of a self-association.
+   * @param connector The connector to normalise
+   * @param dataRegistry The registry to use to extract relevant information from other elements
+   * @returns An array of normalised connectors
+   */
+  public async normalise(
+    connector: EaConnector,
```

Comment:
Not really an issue since it's scoped, but the variable `connector` is already defined on line 29. Might be a bit confusing